### PR TITLE
[2.0.0] Update datacite normalizer

### PR DESCRIPTION
**Pull request by leb2dg**
### Changes
* Update to work with release 2.0.0
* Fix harvester pendulum problem
* Handle both v3 and v4 of Datacite's schema
* Add a few more work relations

### CODE REVIEW

File: providers/org/datacite/normalizer.py [hunk 1]
```diff
@@ -28,243 +74,402 @@ def force_text(data):
...
+    """
+    Returns agent type based on contributor type.
+    """
+    is_not_person = try_contributor_type(agent, NOT_PEOPLE_TYPES)
+    is_person = try_contributor_type(agent, PEOPLE_TYPES)
+
+    if is_person:
+        return 'Person'
+    elif is_not_person:
+        return 'Organization'
```
Comment (aaxelb) [inline providers/org/datacite/normalizer.py]: Could any of the `NOT_PEOPLE_TYPES` be an institution? Not sure if `HostingInstitution` means it the same way we do.

File: providers/org/datacite/normalizer.py [hunk 2]
```diff
@@ -28,243 +74,402 @@ def force_text(data):
...
+    schema = 'Creator'
 
-    name = ctx
+    order_cited = ctx('index')
 
 
-class Identifier(Parser):
-    base_url = tools.Try(ctx['@schemeURI'])
-    url = tools.Join(tools.Concat(tools.Try(ctx['@schemeURI']), tools.Try(ctx['#text'])), joiner='/')
+class HostRelation(Parser):
```
Comment (aaxelb) [inline providers/org/datacite/normalizer.py]: Is `HostRelation` ever used?

File: providers/org/datacite/normalizer.py [hunk 3]
```diff
@@ -28,243 +74,402 @@ def force_text(data):
...
+    name = tools.Try(ctx.awardTitle)
+    description = tools.Try(ctx.awardNumber)
+    uri = tools.Try(ctx.awardURI)
+
+
+class FunderRelation(Parser):
+    schema = 'Funder'
 
+    agent = tools.Delegate(FunderAgent, ctx)
+    award = tools.Delegate(Award, ctx)
```
Comment (aaxelb) [inline providers/org/datacite/normalizer.py]: Should be `awards`, a many-to-many to `Award` through `ThroughAwards`

File: providers/org/datacite/normalizer.py [hunk 4]
```diff
@@ -7,8 +6,55 @@
...
+    'Editor',
+    'ProjectLeader',
+    'ProjectManager',
+    'ProjectMember',
+    'RelatedPerson',
+    'Researcher',
+    'Supervisor',
+    'WorkPackageLeader'
+)
+CHECK_TYPES = (
```
Comment (aaxelb) [inline providers/org/datacite/normalizer.py]: Is `CHECK_TYPES` ever used?

File: providers/org/datacite/normalizer.py [hunk 5]
```diff
@@ -28,243 +74,402 @@ def force_text(data):
...
+        if inverse and relation in INVERSE_RELATIONS:
+            results.append(option)
+        elif not inverse and relation in RELATIONS:
+            results.append(option)
+    return results
+
+
+def get_relation_type(relation_type):
+    normalized_relation = RELATION_MAP[relation_type]
+    return normalized_relation if normalized_relation else 'WorkRelation'
```
Comment (aaxelb) [inline providers/org/datacite/normalizer.py]: Could be simplified:
```
return normalized_relation or 'WorkRelation'
```
I like short circuiting.

File: providers/org/datacite/normalizer.py [hunk 6]
```diff
@@ -28,243 +74,402 @@ def force_text(data):
...
+        tools.Delegate(AgentIdentifier),
+        tools.Try(
+            tools.RunPython(
+                force_text,
+                ctx.nameIdentifier
+            )
+        )
     )
-    type = tools.RunPython('lower', tools.Try(ctx['@relatedIdentifierType']))
+    related_agents = tools.Map(tools.Delegate(AffiliatedAgent), tools.Concat(tools.Try(
```
Comment (aaxelb) [inline providers/org/datacite/normalizer.py]: Needs to go through an agent relation. `IsAffiliatedWith`, maybe?

File: providers/org/datacite/normalizer.py [hunk 7]
```diff
@@ -28,243 +74,402 @@ def force_text(data):
...
-class ContributorOrganization(Parser):
-    schema = 'Organization'
+class PublisherAgent(Parser):
+    schema = tools.GuessAgentType(ctx, default='organization')
 
-    name = ctx.contributorName
+    name = ctx
 
     class Extra:
         contributor_type = tools.Try(ctx.contributorType)
```
Comment (aaxelb) [inline providers/org/datacite/normalizer.py]: `ctx` is the publisher name here, so it'll always be a string, and this `Try` emits a warning for every record. I think you can get rid of it.

File: providers/org/datacite/normalizer.py (hunk 3 above)
Comment (leb2dg) [inline providers/org/datacite/normalizer.py]: This needs to be plural; `award` --> `awards`

**Pull request merged**